### PR TITLE
SLCORE-726 Add debug telemetry option

### PR DESCRIPTION
**Pull request by vojtech-suchy-sonarsource**

### CODE REVIEW

File: backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryHttpClient.java [hunk 1]
```diff
@@ -124,10 +125,19 @@ private void sendDelete(TelemetryPayload payload) {
...
+    }
     try (var response = client.post(endpoint, HttpClient.JSON_CONTENT_TYPE, payload.toJson())) {
       if (!response.isSuccessful() && InternalDebug.isEnabled()) {
         LOG.error("Failed to upload telemetry data: {}", response.toString());
       }
     }
   }
+
+  @VisibleForTesting
+  boolean isDebugTelemetryEnabled(){
```
Comment (serhat-yenican-sonarsource) [inline backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryHttpClient.java:140]: How about `isTelemetryLogEnabled` for the method name and `SONARLINT_TELEMETRY_LOG` for the env var name?

File: rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/initialize/FeatureFlagsDto.java [hunk 2]
```diff
@@ -34,10 +34,11 @@ public class FeatureFlagsDto {
   private final boolean shouldManageServerSentEvents;
   private final boolean enableDataflowBugDetection;
   private final boolean shouldManageFullSynchronization;
+  private final boolean enableTelemetry;
 
 
   public FeatureFlagsDto(boolean shouldManageSmartNotifications, boolean taintVulnerabilitiesEnabled, boolean shouldSynchronizeProjects, boolean shouldManageLocalServer,
-    boolean enableSecurityHotspots, boolean shouldManageServerSentEvents, boolean enableDataflowBugDetection, boolean shouldManageFullSynchronization) {
+    boolean enableSecurityHotspots, boolean shouldManageServerSentEvents, boolean enableDataflowBugDetection, boolean shouldManageFullSynchronization, boolean enableTelemetry) {
```
Comment (serhat-yenican-sonarsource) [inline rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/initialize/FeatureFlagsDto.java:41]: What do you think about creating a default constructor for an instance with default values? Using a builder would even be better, actually.
So we don't need to worry about the usage of IDEs.

File: backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryHttpClient.java (hunk 1 above)
Comment (vojtech-suchy-sonarsource) [reply backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryHttpClient.java]: Addressed as requested

**Review by serhat-yenican-sonarsource — APPROVED** [review]
Validated on Eclipse. Working as expected!
LGTM!

**Pull request merged**